backend/tests/test_api.py

```diff
@@ -2,3 +2,62 @@ def test_hello(client):
     response = client.get("/")
     assert response.status_code == 200
     assert response.json() == {"message": "Hello user! Tip: open /docs or /redoc for documentation"}
+
+
+def test_ping(client):
+    response = client.get("/ping")
+    assert response.status_code == 200
+    assert response.json().get("ping").startswith("pong @")
+
+
+def test_import_from_github(client):
+    response = client.post(
+        "/test/testing_branch/import",
+        json={"description": "test_description"},
+    )
+    assert response.status_code == 200
+    assert response.json() is True
+
+
+def test_upload_taxonomy(client):
+    with open("tests/data/test.txt", "rb") as f:
+        response = client.post(
+            "/test_taxonomy/test_branch/upload",
+            files={"file": f},
+            data={"description": "test_description"},
+        )
+    assert response.status_code == 200
+    assert response.json() is True
+
+
+def test_add_taxonomy_duplicate_branch_name(client):
+    with open("tests/data/test.txt", "rb") as f:
```

Comment:
you don't know the order in which test will be run, you have to import you taxonomy twice in this test. (and that's why we clean db at each test)